[myang2020] iP

myang2020 opened the pull request:
### Duke - Dwight Schrute: A tasks manager for fellow "The Office" Fans

> “Whenever I’m about to do something, I think, ‘Would an idiot do that?’ and if they would, I do not do that thing.” - Dwight Schrute

Duke - Dwight Schrute gives you a peace of mind by helping you keep track of tasks. It is -

- text-based
- complete with easy commands

To download:

download the zip folder from https://github.com/nus-cs2103-AY2324S2/ip.

1. double-click it
2. start using
3. And it is ~~ expensive~~FREE!

Features:

- [X]  Managing tasks 😆
- [ ]  Reminders (coming soon)

If you are a Java programmer, here is the main method
```
    public static void main(String[] args) throws IOException, DukeException {
        new Duke("./data/duke.txt").run();
    }
```

Commit:
Add Gradle support

Eclipse-Dominator committed:
Bump gradle and lib version

Commit:
docs/README.md: Tweak document template

Let's tweak the docs/README.md (which is used as the user guide)
to fit Duke better. Specifically,

1. mention product name in the title
2. mention adding a product screenshot and a product intro
3. tweak the flow to describe feature-by-feature

Commit:
Level-0

Commit:
Echoes user input

Commit:
Stores todos in array

Commit:
Add Task class

Commit:
Added functionality to mark/unmark tasks as done

Commit:
Modified enums

Commit:
Restructure classes to follow OOP principles

Commit:
Complete OOP restructuring

Commit:
Add testing

Commit:
Added exceptions for incorrect user input

Commit:
Add function to delete events

Commit:
Add FileHandler class and store todo task data in external text file

Commit:
Add comments

Commit:
Merge branch 'branch-Level-7'

Commit:
Add comments

Commit:
Add date format for deadline tasks

Commit:
Fix bug

Commit:
Rename FileHandler to Storage

Commit:
Adhere to OOP style and add relevant classes

Commit:
Organize packages

Commit:
Commit changes in duke.txt file

Commit:
Merge branch 'branch-A-Packages'

* branch-A-Packages:
  Organize packages

Commit:
Merge commit 'a6f73244a20539e20c2189973ebf597df34ef631'

* commit 'a6f73244a20539e20c2189973ebf597df34ef631':
  Bump gradle and lib version
  Add Gradle support
  docs/README.md: Tweak document template

Commit:
Create gradle

Commit:
Add comments to DukeException.java file

Commit:
Merge branch 'branch-A-Gradle'

* branch-A-Gradle:
  Create gradle

Commit:
Add JUnit tests for TodoTask and Parser classes

Commit:
Add method comments

Commit:
Merge branch 'A-JUnit'

* A-JUnit:
  Add JUnit tests for TodoTask and Parser classes

Commit:
Add Javadoc comments

Commit:
Ensure compliance with Java coding standards

Commit:
Implement find function

Commit:
no message

Commit:
Merge branch 'branch-Level-9'

* branch-Level-9:
  Implement find function

Commit:
Merge branch 'branch-A-CodingStandard'

* branch-A-CodingStandard:
  Ensure compliance with Java coding standards

Commit:
Merge branch 'branch-A-JavaDoc'

* branch-A-JavaDoc:
  Add Javadoc comments

Commit:
Run checkstyle files

Commit:
Adhere to java style

Commit:
Merge branch 'branch-A-CheckStyle'

* branch-A-CheckStyle:
  Run checkstyle files

Apzure commented on src/main/java/duke/util/Ui.java:
```diff
@@ -0,0 +1,56 @@
+     * Prints welcome message
+     */
+    public void welcome() {
+        String chat_name = "Dwight Schrute";
```
Intimidating name for chatbot

Apzure commented on src/main/java/duke/Duke.java:
```diff
@@ -0,0 +1,168 @@
+                            break;
+                        }
+
+                    case("delete"):
+                        try {
+                            idx = Parser.parse_delete(input);
+                            System.out.println("\tNoted. I've removed this task:");
+                            System.out.printf("\t\t%s\n", todo.deleteTask(idx));
+                            System.out.printf("\tNow you have %d tasks in the list.\n", todo.size());
+                            break;
+                        } catch (DukeException err) {
+                            System.out.println(err.getMessage());
+                            break;
+                        }
+                    case("find"):
+                        try {
+                            keyword = Parser.parse_find(input);
+                            System.out.println("\tHere are the matching tasks in your list:");
+                            todo.find(keyword);
+                            break;
+                        } catch (DukeException err) {
+                            System.out.println(err.getMessage());
+                            break;
+                        }
+
+                    default:
+                        throw new DukeException("\tSorry, I did not understand the command!");
+
+                }
+                System.out.printf("\t%s\n", divider);
+                storage.writeFile(todo.getList());
+            } catch (DukeException err) {
+                System.out.println(err.getMessage());
+                System.out.printf("\t%s\n", divider);
+            } catch (IOException err) {
+                System.out.println(err.getMessage());
+                System.out.printf("\t%s\n", divider);
+            }
+        }
+    }
```
1. It might be better to have a separate class to handle commands.
2. Abstract out the printing of tasks-related messages to the methods inside Task.java themselves.

Apzure commented on src/main/java/duke/task/EventTask.java:
```diff
@@ -0,0 +1,43 @@
+     * @return String representation of EventTask
+     */
+    @Override
+    public String toString() {
```
Implement the toString() method in your Task.java.
This way you can call it by super.toString() in each of the subclasses of Task.java.
This is helpful since each of the subclasses toString() methods have commonalities in the string it prints.
For reference, this is mine:
 String check = isDone ? "[X] " : "[ ] ";
 return check + this.name;

Apzure commented on src/main/java/duke/util/Storage.java:
```diff
@@ -0,0 +1,92 @@
+                        }
+
+                    case ("deadline"):
+                        first_string = input.split(" /")[0];
+                        second_string = input.split(" /")[1];
+                        task_name = String.join(" ", Arrays.copyOfRange(first_string.split(" "), 1, first_string.split(" ").length));
+                        DateTimeFormatter formatter = DateTimeFormatter.ofPattern("d/M/yyyy HHmm");
+                        end = String.join(" ", Arrays.copyOfRange(second_string.split("by "), 1, second_string.split("by ").length));
+                        LocalDateTime end_time = LocalDateTime.parse(end, formatter);
```
More readable if you abstract this into a method

Apzure reviewed (commented):
LGTM! Keep up the good work!
You can consider some of the changes I recommended!

Commit:
Set up introduction part of JavaFX tutorial

Commit:
Implement GUI that echoes user input

Commit:
Complete transfer of CLI to JavaFx GUI

Commit:
no message

Commit:
Merge branch 'branch-Level-10'

* branch-Level-10:
  Complete transfer of CLI to JavaFx GUI
  Implement GUI that echoes user input
  Set up introduction part of JavaFX tutorial

# Conflicts:
#	src/main/java/duke/Duke.java

Commit:
Delete obsolete functions

Commit:
Fix bugs

Commit:
Add assertions

Commit:
Ensure code quality

myang2020 committed:
Merge pull request #3 from myang2020/branch-A-Assertions

Add assertions

myang2020 committed:
Merge pull request #2 from myang2020/branch-A-CodeQuality

Branch a code quality

Commit:
Merge branch 'master' of https://github.com/myang2020/ip

* 'master' of https://github.com/myang2020/ip:
  Ensure code quality
  Add assertions

Commit:
Ensure code quality for variable names

Parser class methods: variables do not follow camel case

Variables in parser class methods have to be renamed

This will ensure code quality

Commit:
Enable flexible task searches

Commit:
Add image of UI

Commit:
Modify based on SLAP

Commit:
Rename project

Commit:
Add ReadME

Commit:
Fix bugs

Commit:
Fix jar settings

Commit:
Create directory to save data
